lib/tests/streamlit/runtime/secrets_test.py

```diff
@@ -116,6 +118,24 @@ def test_getattr_nonexistent(self, *mocks):
         with self.assertRaises(AttributeError):
             self.secrets.subsection.nonexistent_secret
 
+    @patch("streamlit.watcher.path_watcher.watch_file")
+    @patch("builtins.open", new_callable=mock_open, read_data=MOCK_TOML)
+    def test_getattr_raises_exception_on_attr_dict(self, *mocks):
+        """Verify that assignment to nested secrets raises TypeError."""
+        with self.assertRaises(TypeError):
+            self.secrets.subsection["new_secret"] = "123"
+
+        with self.assertRaises(TypeError):
+            self.secrets.subsection.new_secret = "123"
+
+    @patch("streamlit.watcher.path_watcher.watch_file")
+    @patch("builtins.open", new_callable=mock_open, read_data=MOCK_TOML)
+    def test_attr_dict_is_mapping_but_not_built_in_dict(self, *mocks):
+        """Verify that AttrDict implements Mapping, but not built-in Dict"""
+        self.assertTrue(isinstance(self.secrets.subsection, Mapping))
```

Comment:
nit: Maybe also check that it's not a `MutableMapping`

Reply:
Done!